src/python/pants/option/global_options.py

```diff
@@ -355,6 +358,10 @@ def register_bootstrap_options(cls, register):
              # This default is also hard-coded into the engine's rust code in
              # fs::Store::default_path
              default=os.path.expanduser('~/.cache/pants/lmdb_store'))
+
+    register('--enable-remoting', advanced=True, type=bool,
```

Comment:
We should ensure that  --remote-execution-server exists if --enable-remoting is passed.

Reply:
Question about that. I was going to but then saw that we have validation in `scheduler.py` about `remote_execution_server` and `remote_store_server`:

https://github.com/pantsbuild/pants/blob/8054995ef91d233a5e1d660e87304a32d0adeb65/src/python/pants/engine/scheduler.py#L78-L79

1) Should that validation be moved into `global_options`'s `validate_instance()`? Seems like a good idea to centralize validation.
2) How does `--remote-store-server` work? That is, would there ever be a situation where users want to use the remote store but not remote execution?

Reply:
>Should that validation be moved into global_options's validate_instance()? Seems like a good idea to centralize validation.

That seems like a good idea to me.

>How does --remote-store-server work? That is, would there ever be a situation where users want to use the remote store but not remote execution?

I don't think so. The idea of the store is to upload artifacts before making an `execute` RPC call, which only has file digests in it, that reference blobs in the store.

Reply:
Got it. Thank you!